modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/imageio/MaskOverviewProvider.java

```diff
@@ -97,62 +97,74 @@
     private ImageInputStreamSpi maskOvrStreamSpi;
 
     private ImageReaderSpi maskOvrReaderSpi;
-
+    
     public MaskOverviewProvider(DatasetLayout layout, File inputFile) throws IOException {
         this(layout, inputFile, (ImageReaderSpi) null);
     }
 
     public MaskOverviewProvider(DatasetLayout layout, File inputFile, ImageReaderSpi suggestedSPI)
             throws IOException {
-        this (layout, inputFile, new SpiHelper(inputFile, suggestedSPI));
+        this (layout, inputFile, new SpiHelper(inputFile.toURI().toURL(), suggestedSPI));
+    }
+
+    public MaskOverviewProvider(DatasetLayout layout, URL inputFile) throws IOException {
+        this(layout, inputFile, (ImageReaderSpi) null);
     }
 
+    public MaskOverviewProvider(DatasetLayout layout, URL inputFile, ImageReaderSpi suggestedSPI)
+            throws IOException {
+        this (layout, inputFile, new SpiHelper(inputFile, suggestedSPI));
+    }
+    
     public MaskOverviewProvider(DatasetLayout layout, File inputFile, SpiHelper spiProvider) throws IOException {
+        this(layout, inputFile.toURI().toURL(), spiProvider);
+        
+    }
+
+    public MaskOverviewProvider(DatasetLayout layout, URL inputFile, SpiHelper spiProvider) throws IOException {
         ImageReaderSpi suggestedSPI = spiProvider.getSuggestedSpi();
+        ImageInputStreamSpi suggestedStreamSPI = spiProvider.getSuggestedStreamSpi();
         readerSpi = spiProvider.getReaderSpi();
         streamSpi = spiProvider.getStreamSpi();
         this.fileURL = spiProvider.getFileURL();
         this.layout = layout;
 
         // Handling Overviews
-        File overviewFile = new File(inputFile.getAbsolutePath() + OVR_EXTENSION);
+        ovrURL = new URL(inputFile.toString() + OVR_EXTENSION);
         hasDatasetLayout = layout != null;
         if (hasDatasetLayout && layout.getExternalOverviews() != null) {
-            overviewFile = layout.getExternalOverviews();
+            ovrURL = layout.getExternalOverviews().toURI().toURL();
         }
-        if (overviewFile.exists() && overviewFile.canRead()) {
-            // Creating overview file URL
-            ovrURL = URLs.fileToUrl(overviewFile);
-            // Creating cached SPIs
-            overviewStreamSpi = getInputStreamSPIFromURL(ovrURL);
-            ImageInputStream ovrStream = null;
-            try {
-                ovrStream = overviewStreamSpi.createInputStreamInstance(ovrURL,
-                        ImageIO.getUseCache(), ImageIO.getCacheDirectory());
+        // Creating overview file URL
+        // ovrURL = URLs.fileToUrl(overviewFile);
```

Comment:
Don't leave commented out sections in the code. Not sure if there are others, please have a look.